lib/pbench/server/api/resources/endpoint_configure.py

```diff
@@ -0,0 +1,145 @@
+            queryMonthIndices: Return the YYYY-mm date strings for all ES
+                vx.run-data.* indices, in descending order
+
+        TODO: We need an internal mechanism to track the active versions of the
+        various Elasticsearch template documents. We're hardcoding them here and
+        in other APIs. We should consider persisting an equivalent of the
+        mapping table built in "indexer.py" for use across the server APIs.
+
+        TODO: We provide Elasticsearch index root names here, which the dashboard
+        code needs to perform the queries we've not yet replaced with server-side
+        implementations. The entire "indices" section can be removed once that is
+        resolved.
+        """
+        self.logger.info("Received these headers: {!r}", request.headers)
+        header = request.headers.get("Forwarded")
+        origin = None
+        if header:
+            m = self.forward_pattern.search(header)
+            if m:
+                origin = m.group("host")
+                self.logger.info("Forwarded: {}, ({})", header, origin)
+        if not origin:
+            header = request.headers.get("X-Forwarded-Host")
+            if header:
+                m = self.x_forward_pattern.search(header)
+                if m:
+                    origin = m.group("host")
+                    self.logger.info("X-Forwarded-Host: {}, ({})", header, origin)
+        if not origin:
+            origin = self.host
+        host = f"http://{origin}"
+        uri = urljoin(host, self.uri_prefix)
+        self.logger.info("'{}' : '{}' : '{}' : '{}'", self.host, origin, host, uri)
+
+        # Strip a trailing slash because the paths we'll add have an
+        # initial slash and we don't want two.
+        if uri.endswith("/"):
+            uri = uri.split(0, -1)
+
+        try:
+            endpoints = {
+                "identification": f"Pbench server {self.commit_id}",
+                "indices": {
+                    "run_index": f"{self.prefix}.v6.run-data.",
+                    "run_toc_index": f"{self.prefix}.v6.run-toc.",
+                    "result_index": f"{self.prefix}.v5.result-data-sample.",
+                    "result_data_index": f"{self.prefix}.v5.result-data.",
+                },
+                "api": {
+                    "results": f"{host}/results",
+                    "elasticsearch": f"{uri}/elasticsearch",
+                    "endpoints": f"{uri}/endpoints",
+                    "graphql": f"{uri}/graphql",
+                    "controller_list": f"{uri}/controllers/list",
+                    "controller_months": f"{uri}/controllers/months",
+                    "dataset_list": f"{uri}/datasets/list",
+                    "dataset": f"{uri}/datasets",
+                    "register": f"{uri}/register",
+                    "login": f"{uri}/login",
+                    "logout": f"{uri}/logout",
+                    "user": f"{uri}/user/<string:username>",
```

Comment:
This is cool, but doesn't this mean that if a new endpoint is added, it needs to be manually added here?  I wonder if there is a way to dynamically derive this list from Flask somehow?  Perhaps in another PR?

Reply:
Yes. In theory I think it's possible to traverse the endpoints in Flask. I started using the names of the existing dashboard service methods, which couldn't be derived. Now... well, it's an interesting idea. After Webb pushed me far enough past the tipping point, I renamed them to be better aligned with the endpoints. Aside from a couple of minor variations, I would have just taken the URI path past the `/api/v1/` and replaced the slashes with "_", so it's not impossible. Maybe I'll even give it a try...